TRAMS-API/Repositories/AcademiesRepository.cs

```diff
@@ -0,0 +1,72 @@
+﻿using API.HttpHelpers;
+using API.Mapping;
+using API.Models.D365;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace API.Repositories
+{
+    public class AcademiesRepository
+    {
+        private static readonly string _route = "accounts";
+
+        private readonly AuthenticatedHttpClient _client;
+        private readonly IODataModelHelper<GetAcademiesD365Model> _oDataHelper;
+
+        public AcademiesRepository(AuthenticatedHttpClient client, IODataModelHelper<GetAcademiesD365Model> oDataHelper)
+        {
+            _client = client;
+            _oDataHelper = oDataHelper;
+        }
+
+        public async Task<GetAcademiesD365Model> GetAcademyById(Guid id)
+        {
+            var fields = _oDataHelper.GetSelectClause();
+            var expandClause = _oDataHelper.GetExpandClause();
+
+            var url = ODataUrlBuilder.BuildRetrieveOneUrl(_route, id, fields, expandClause);
```

Comment:
Suggestion: I think it would be super useful to make ODataUrlBuilder an open generic and say ODataUrlBuilder<GetAcademiesD365Model>.BuildRetrieveOneUrl... etc and the internals called GetSelectClause and GetExpandClause rather than having to proceed this with those two calls.  Might cause some issues with getting the appropriate IODataModelHelper so not sure how viable it would be without some refactoring.

Reply:
Perfect idea, thanks!